File: Application/DTO/Response/ShopResponsd.cs
using Queue.Domain.Entities;

public class ShopResponse
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string PublicSlug { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public int Status { get; set; } = 0;
    public int OwnerId { get; set; } = 0;
    public List<ShopBusinessHour>? ShopHours { get; set; } = new List<ShopBusinessHour>();
    public List<ShopHoliday>? ShopHolidays { get; set; } = new List<ShopHoliday>();
    public List<BranchDto>? ShopBranches { get; set; } = new List<BranchDto>();
    public List<Service>? Services { get; set; } = new List<Service>();
    public bool IsActive { get; set; } = true;
    public string? Logo { get; set; }
    public string? Description { get; set; }
    public string? Email { get; set; }
}

public class BranchDto
{
    public int Id { get; set; }
    public Guid Guid { get; set; }
    public Guid PublicBookingId { get; set; }
    public bool IsOnlineBookingEnabled { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public AddressDto? Address { get; set; }
    public List<ShopBusinessHour>? BusinessHours { get; set; } = new List<ShopBusinessHour>();
}

public class AddressDto
{
    public string HouseNo { get; set; } = string.Empty;
    public string Street { get; set; } = string.Empty;
    public string Subdistrict { get; set; } = string.Empty;
    public string District { get; set; } = string.Empty;
    public string Province { get; set; } = string.Empty;
    public string Zipcode { get; set; } = string.Empty;
    public string FullAddress =>
        $"{HouseNo} {Street} {Subdistrict} {District} {Province} {Zipcode}".Trim();
}

File: Frontend/src/types/shop/shop-responsd.ts
import { SetService } from "./service";

export interface ShopResponse {
  id: number;
  name: string;
  publicSlug: string;
  type: string;
  address: string;
  phone: string;
  status: number;
  ownerId: number;
  shopHours: ShopBusinessHour[];
  shopHolidays: ShopHoliday[];
  shopBranches: BranchDto[];
  services: SetService[];
  logo : string;
  isActive : boolean;
  description?: string;
  email?: string;
}

export interface ShopBusinessHour {
  shopId: number;
  dayOfWeek: number;
  isOpen: boolean;
  openTime: string;
  closeTime: string;
}

export interface ShopHoliday {
  shopId: number;
  date: string;
  description?: string;
}

export interface BranchDto {
  id: number;
  guid: string;
  publicBookingId: string;
  isOnlineBookingEnabled: boolean;
  name: string;
  phone: string;
  address?: AddressDto;
  isSelected?: boolean;
}

export interface AddressDto {
  houseNo: string;
  street: string;
  subdistrict: string;
  district: string;
  province: string;
  zipcode: string;
  fullAddress: string;
}
